### PR TITLE
Fix sorting (order by in inner select is not preserved after joining)

### PR DESCRIPTION
**Pull request by datazuul**
* Fix sorting (order by in inner select is not preserved after joining in outer select!). add order by to outer select, too
* in articlerepositoryimpl the method getCreators was fixed
* get random fixed (returning list instead pageresponse now)

### CODE REVIEW

File: dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/IdentifiableRepositoryImpl.java [hunk 1]
```diff
@@ -527,22 +537,19 @@ public long retrieveCount(StringBuilder sqlCount, final Map<String, Object> argu
             + ".uuid = id.identifiable"
             + " LEFT JOIN fileresources_image AS file ON "
             + tableAlias
-            + ".previewfileresource = file.uuid";
+            + ".previewfileresource = file.uuid"
+            + (orderBy != null ? orderBy : "");
```
Comment (morpheus-87) [inline dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/IdentifiableRepositoryImpl.java:541]:
```suggestion
            + (orderBy != null ? " " + orderBy : "");
```

File: dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/ArticleRepositoryImpl.java [hunk 2]
```diff
@@ -82,13 +76,12 @@ public static String getSqlUpdateFieldValues() {
         + ", date_published=:datePublished, text=:text::JSONB, timevalue_published=:timeValuePublished::JSONB";
   }
 
-  private final EntityRepositoryImpl<EntityImpl> entityRepositoryImpl;
+  @Autowired
+  @Qualifier("entityRepositoryImpl")
+  private EntityRepositoryImpl<EntityImpl> entityRepositoryImpl;
```
Comment (morpheus-87) [inline dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/ArticleRepositoryImpl.java:81]: Besser im Konstruktor?

File: dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/EntityRepositoryImpl.java [hunk 3]
```diff
@@ -215,14 +209,13 @@ public E findOneByRefId(long refId) {
...
+  public List<E> findRandom(int count) {
     // Warning: could be very slow if random is used on tables with many million records
     // see https://www.gab.lc/articles/bigdata_postgresql_order_by_random/
     StringBuilder innerQuery =
         new StringBuilder("SELECT * FROM " + tableName + " ORDER BY RANDOM() LIMIT " + count);
-    List<E> randomList = retrieveList(sqlSelectReducedFields, innerQuery, null);
-    PageRequest pageRequest = new PageRequestImpl(0, count);
-    return new PageResponseImpl<>(randomList, pageRequest, count);
+    List<E> randomList = retrieveList(sqlSelectReducedFields, innerQuery, null, null);
+    return randomList;
```
Comment (morpheus-87) [inline dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/EntityRepositoryImpl.java:218]: Besser direkt `return`.

**Review by morpheus-87 — CHANGES_REQUESTED** [review]

File: dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/EntityRepositoryImpl.java (hunk 3 above)
Comment (datazuul) [reply dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/EntityRepositoryImpl.java]: done

File: dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/ArticleRepositoryImpl.java (hunk 2 above)
Comment (datazuul) [reply dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/identifiable/entity/ArticleRepositoryImpl.java]: done

**Review by morpheus-87 — APPROVED** [review]

**Pull request merged**